File: legacy_perl/translate_orf.pl
#!/usr/bin/env perl
# Translate ORFs from a FASTA file (simple approach)
use strict;
use warnings;
use Bio::SeqIO;
use Bio::Tools::CodonTable;
my $in = shift or die "Usage: $0 input.fasta\n";
my $seqio = Bio::SeqIO->new(-file=>$in, -format=>'fasta');
my $table = Bio::Tools::CodonTable->new(-id=>1);
while(my $r = $seqio->next_seq){
    my $seq = $r->seq;
    for(my $i=0;$i<3;$i++){
        my $pep = $table->translate($seq, $i, undef, 1);
        print ">", $r->display_id, "_frame", $i+1, "\n", $pep, "\n";
    }
}

File: biopython_rewrite/translate_orf.py
#!/usr/bin/env python3
# Translate ORFs from FASTA (simple)
from Bio import SeqIO
from Bio.Seq import Seq
import sys
if len(sys.argv)<2:
    print('Usage: translate_orf.py input.fasta'); sys.exit(1)
for rec in SeqIO.parse(sys.argv[1],'fasta'):
    seq = rec.seq
    for frame in range(3):
        pep = seq[frame:].translate(to_stop=False)
        print(f">{rec.id}_frame{frame+1}\n{pep}")
